### PR TITLE
Edit disk form components

### LINKED ISSUE #999: Form validation and input components: settings -> edit disk

### LINKED ISSUE #1057: Edit disk isn't rendering properly
screen shot 2017-01-20 at 11 28 36 am

### PR DESCRIPTION
**Pull request by abemassry**
closes #999 
closes #1057 

image

image

### CODE REVIEW

**Comment by eatonphil** [comment]
Also closes #1057?

File: src/linodes/linode/settings/components/EditModal.js [hunk 1]
```diff
@@ -50,41 +52,55 @@ export class EditModal extends Component {
...
+          <div className="col-sm-5 label-col">
+            <label htmlFor="size">New size (MB)</label>
+          </div>
+          <div className="col-sm-7">
+            <Input
+              type="number"
+              min={8} /* TODO: can't/don't calculate distro min size requirement */
+              max={free + disk.size}
+              value={size}
+              name="size"
```
Comment (eatonphil) [inline src/linodes/linode/settings/components/EditModal.js]: I think you're missing `id="size"` here

File: src/linodes/linode/settings/components/EditModal.js [hunk 2]
```diff
@@ -50,41 +52,55 @@ export class EditModal extends Component {
...
-            onChange={e => {
-              this.setState({ size: parseInt(e.target.value, 10) });
-            }}
-          />
+        <FormGroup errors={errors} name="filesystem" className="row">
+          <div className="col-sm-5 label-col">
+            <label>Format</label>
+          </div>
+          <div className="col-sm-7">
+            <input disabled className="form-control" value={disk.filesystem} />
```
Comment (eatonphil) [inline src/linodes/linode/settings/components/EditModal.js]: Replace this with `Input`? (And on line 84)

File: src/linodes/linode/settings/components/EditModal.js [hunk 3]
```diff
@@ -50,41 +52,55 @@ export class EditModal extends Component {
...
-            placeholder="Label"
-            value={label}
-            disabled={loading}
-            onChange={e => {
-              this.setState({ label: e.target.value });
-            }}
-          />
+        <FormGroup errors={errors} name="label" className="row">
+          <div className="col-sm-5 label-col">
+            <label>Label</label>
```
Comment (eatonphil) [inline src/linodes/linode/settings/components/EditModal.js]: htmlFor?

File: src/linodes/linode/settings/components/EditModal.js (hunk 1 above)
Comment (eatonphil) [inline src/linodes/linode/settings/components/EditModal.js]: (And you can drop the name)

**Comment by briansteffens** [comment]
Not a bad PR, although I feel it could use an assembler, perhaps made in Haskell and nowhere near complete, if only we knew where to find something like that...

**Comment by abemassry** [comment]
@eatonphil yes it closes that one as well

**Pull request merged**